### PR TITLE
Fix specs Makefile

### PR DESCRIPTION
**Pull request by msprotz**
Santiago, Karthik recently reported that the specs local Makefile no longer works. I tried to patch it up, but I thought I'd run the changes by you since you touched it recently.

Some relative paths seemed to need a fix, and somehow that shortest stem rule didn't seem to kick in for the .exe rule.

We should also enforce that this local Makefile uses a recent GNU make.

### CODE REVIEW

File: specs/Makefile [hunk 1]
```diff
@@ -11,14 +11,16 @@ FSTAR_ROOTS += $(wildcard tests/*.fst) $(wildcard alternative/*.fst) $(wildcard
...
 
-.PRECIOUS: specs/tests/%_AutoTest.ml
-specs/tests/%_AutoTest.ml:
+.PRECIOUS: tests/%_AutoTest.ml
+tests/%_AutoTest.ml:
 	echo "if not ($*.test ()) then (print_endline \"$* failed\"; exit 1)" > $@
 
-.PRECIOUS: specs/tests/%.exe
-specs/tests/%.exe: $(ALL_CMX_FILES) specs/tests/%_AutoTest.ml
+%.exe:
```
Comment (msprotz) [inline specs/Makefile]: note: this cancels the previous pattern rule now defined in Makefile.local

**Comment by s-zanella** [comment]
This looks good to me, but I'm curious how it was working for me before. 
Let me try locally before merging.

**Comment by msprotz** [comment]
Thanks Santiago. I wonder if we should do something like this in the nightly build

```
for f in code/*/Makefile specs/Makefile lib/Makefile; do
  git clean -fdx && (cd $(dirname $f) && make -j)
done
```

thoughts?

**Comment by s-zanella** [comment]
It looks expensive to do a `git clean -fdx` after each build. Leaving out the `git clean -fdx &&` part would be a cheap sanity test.

**Pull request merged**